pkg/referrerstore/oras/oras.go

```diff
@@ -336,6 +336,12 @@ func (store *orasStore) createRepository(ctx context.Context, targetRef common.R
 
 	// set the provider to return the resolved credentials
 	credentialProvider := func(ctx context.Context, registry string) (auth.Credential, error) {
+		if authConfig.Username != "" && authConfig.IdentityToken != "" {
+			return auth.Credential{
```

Comment:
[auth.Credential](https://pkg.go.dev/oras.land/oras-go/v2@v2.0.0-rc.5/registry/remote/auth#Credential) supports identity tokens natively.

```go
// RefreshToken is a bearer token to be sent to the authorization service
// for fetching access tokens.
// A refresh token is often referred as an identity token.
// Reference: https://docs.docker.com/registry/spec/auth/oauth/
RefreshToken string
```

Reply:
Updated to pass `RefreshToken` to ORAS and letting it handle the logic vs having separate conditions in Ratify

Validated that it works as expected for both `Username`/`Password` and for `RefreshToken`